### PR TITLE
Rename extension from `suse-jeos` to `suse-chost`

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area os
/kind cleanup
/priority normal
/os suse-chost

**What this PR does / why we need it**:
With this pull request we are going to rename the extension from `suse-jeos` to `suse-chost`.

**Special notes for your reviewer**:
* ✅ Depends on #19

/assign @vpnachev
/invite @vpnachev 

**Release note**:

```action operator
The SuSE JeOS extension controller has now been renamed to SuSE CHost. It is still responsible for both extension types (`suse-chost` and `suse-jeos`, though, `suse-jeos` is deprecated and might be removed in a future version). Please adapt your `ControllerRegistration` resources accordingly. If you currently run the SuSE JeOS extension in your Gardener system you have to `kubectl delete` the respective `ControllerRegistration` as well as all `ControllerInstallation`s before you can apply the new `ControllerRegistration` resource for the SuSE CHost extension.
```

### CODE REVIEW

**Comment by vpnachev** [comment]
The deletion of the old `controllerregistration` is not allowed as long as there is at least one shoot using it, e.g. there are `controllerinstallation` resources. 

However, if all `controllerinstallation` resources are deleted, then the `controllerregistration` can be deleted as well, so please add a release note that all controller installations `os-suse-jeos-*` must be manually deleted after the new controller registration is created.


Will you also rename the repo ` gardener-extension-os-suse-jeos` -> `gardener-extension-os-suse-chost`  ?

**Comment by rfranzke** [comment]
Thanks @vpnachev, I've added the release note.
/ready-for-review

Yes, after the PR has been merged I will rename the repository and I will adapt the CI pipeline definition in the `refs/meta/ci`. This means, however, that this PR won't be able to build the Docker image.

**Comment by rfranzke** [comment]
/unhold
@vpnachev everything okay now?

File: README.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
+[![Go Report Card](https://goreportcard.com/badge/github.com/gardener/gardener-extension-os-suse-chost)](https://goreportcard.com/report/github.com/gardener/gardener-extension-os-suse-chost)
 
-This controller operates on the [`OperatingSystemConfig`](https://github.com/gardener/gardener/blob/master/docs/proposals/01-extensibility.md#cloud-config-user-data-for-bootstrapping-machines) resource in the `extensions.gardener.cloud/v1alpha1` API group. It manages those objects that are requesting [SUSE JeOS](https://www.suse.com/products/server/jeos/) configuration and supports both types: (`.spec.type=suse-jeos`): and (`.spec.type=suse-chost`):
+This controller operates on the [`OperatingSystemConfig`](https://github.com/gardener/gardener/blob/master/docs/proposals/01-extensibility.md#cloud-config-user-data-for-bootstrapping-machines) resource in the `extensions.gardener.cloud/v1alpha1` API group. It manages those objects that are requesting [SUSE Container Host](https://www.suse.com/products/server/jeos/) configuration and supports both types (`.spec.type=suse-chost` and `.spec.type=suse-jeos`):
```
Comment (vpnachev) [inline README.md:6]: As we know chost is not jeos, so you can remove the link below.
```suggestion
This controller operates on the [`OperatingSystemConfig`](https://github.com/gardener/gardener/blob/master/docs/proposals/01-extensibility.md#cloud-config-user-data-for-bootstrapping-machines) resource in the `extensions.gardener.cloud/v1alpha1` API group. It manages those objects that are requesting SUSE Container Host configuration and supports both types (`.spec.type=suse-chost` and `.spec.type=suse-jeos`):
```

File: charts/gardener-extension-os-suse-chost/doc.go [hunk 2]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:generate ../../vendor/github.com/gardener/gardener/extensions/hack/generate-controller-registration.sh os-suse-jeos . ../../VERSION ../../example/controller-registration.yaml OperatingSystemConfig:suse-jeos OperatingSystemConfig:suse-chost
+//go:generate ../../vendor/github.com/gardener/gardener/extensions/hack/generate-controller-registration.sh os-suse-chost . ../../VERSION ../../example/controller-registration.yaml OperatingSystemConfig:suse-chost OperatingSystemConfig:suse-jeos
```
Comment (vpnachev) [inline charts/gardener-extension-os-suse-chost/doc.go:15]: Not at the scope of this PR, but it us very little change.

```suggestion
//go:generate ../../vendor/github.com/gardener/gardener/extensions/hack/generate-controller-registration.sh os-suse-chost . ../../VERSION ../../example/controller-registration.yaml OperatingSystemConfig:suse-chost OperatingSystemConfig:suse-jeos OperatingSystemConfig:memoryone-chost
```

PS. you have to run `make generate`

**Pull request merged**